staging/src/k8s.io/client-go/util/jsonpath/jsonpath_test.go

```diff
@@ -806,8 +807,8 @@ func TestRunningPodsJSONPathOutput(t *testing.T) {
 	testJSONPath(
 		[]jsonpathTest{
 			{
-				"when range is used in a certain way in script, additional line is printed",
-				`{range .items[?(.status.phase=="Running")]}{.metadata.name}{" is Running\n"}`,
+				"range over pods without selecting the last one",
+				`{range .items[?(.status.phase=="Running")]}{.metadata.name}{" is Running\n"}{end}`,
```

Comment:
shouldn't we have both cases tested, ie with and without end?

Reply:
Should we test invalid syntaxes here? What should be the behaviour?
- the output is broken and outputs additional line (current behaviour)
- or should we fast fail with invalid syntax error? (would require changes)

Reply:
it was kind of graceful before to leave out the `{end}`, just a little wrong. Could be get it working easily too?

Reply:
In any case, I would rather try to avoid breaking behaviour that ran without error before. Fixing it is ok, but not suddenly fail.

Reply:
I am not sure how easy that would be. 
But do we want to support an invalid syntax? I think it is a bit problematic of supporting this especially when you consider nested ranges. I would prefer an error here , but that would also break some scripts (more than keeping it failing slightly).

Reply:
I think that the support for missing `{end}` was rather accidental than intentional. I'd prefer we stick with having code simple and complying to jsonpath standard, than over-complicated to support something not entirely correct.

Reply:
I would prefer keeping graceful behaviour and not fix it and break users. But I am not blocking this. Both paths are acceptable.